be/src/udf/java/java_udf.h

```diff
@@ -0,0 +1,279 @@
+    std::string to_cxx_string(jstring str);
+    std::string dumpExceptionString(jthrowable throwable);
+    jmethodID getToStringMethod(jclass clazz);
+    jstring to_jstring(const std::string& str);
+    jmethodID getMethod(jclass clazz, const std::string& method, const std::string& sig);
+    jmethodID getStaticMethod(jclass clazz, const std::string& method, const std::string& sig);
+
+    DECLARE_NEW_BOX(uint8_t, Boolean)
+    DECLARE_NEW_BOX(int8_t, Byte)
+    DECLARE_NEW_BOX(int16_t, Short)
+    DECLARE_NEW_BOX(int32_t, Integer)
+    DECLARE_NEW_BOX(int64_t, Long)
+    DECLARE_NEW_BOX(float, Float)
+    DECLARE_NEW_BOX(double, Double)
+
+    jobject newString(const char* data, size_t size);
+
+    int intVal(jobject jinteger);
+    Slice sliceVal(jstring jstr);
+    size_t string_length(jstring jstr);
+    Slice sliceVal(jstring jstr, std::string* buffer);
+    // replace '.' as '/'
+    // eg: java.lang.Integer -> java/lang/Integer
+    static std::string to_jni_class_name(const std::string& name);
+
+private:
+    JVMFunctionHelper(JNIEnv* env) : _env(env) {}
+    void _init();
+    void _add_class_path(const std::string& path);
+
+private:
+    JNIEnv* _env;
+
+    DEFINE_JAVA_PRIM_TYPE(boolean);
+    DEFINE_JAVA_PRIM_TYPE(byte);
+    DEFINE_JAVA_PRIM_TYPE(short);
+    DEFINE_JAVA_PRIM_TYPE(int);
+    DEFINE_JAVA_PRIM_TYPE(long);
+    DEFINE_JAVA_PRIM_TYPE(float);
+    DEFINE_JAVA_PRIM_TYPE(double);
+
+    jclass _object_class;
+    jclass _string_class;
+    jclass _throwable_class;
+    jclass _jarrays_class;
+
+    jmethodID _string_construct_with_bytes;
+
+    jobject _utf8_charsets;
+};
+
+// Object Guard
+// auto release jobject
+class ObjectGuard {
+public:
+    ObjectGuard(jobject&& obj) {}
+    ~ObjectGuard() noexcept;
+
+private:
+    jobject _obj;
+};
+
+class DirectByteBuffer {
+public:
+    static constexpr const char* JNI_CLASS_NAME = "java/nio/ByteBuffer";
+
+    DirectByteBuffer(void* data, int capacity);
+    DirectByteBuffer(jobject&& handle, void* data, int capacity)
+            : _handle(std::move(handle)), _data(data), _capacity(capacity) {}
+    ~DirectByteBuffer();
+
+    DirectByteBuffer(const DirectByteBuffer&) = delete;
+    DirectByteBuffer& operator=(const DirectByteBuffer& other) = delete;
+
+    DirectByteBuffer(DirectByteBuffer&& other) {
+        _handle = other._handle;
+        _data = other._data;
+        _capacity = other._capacity;
+
+        other._handle = nullptr;
+    }
+
+    DirectByteBuffer& operator=(DirectByteBuffer&& other) {
+        DirectByteBuffer tmp(std::move(other));
+        std::swap(this->_handle, tmp._handle);
+        std::swap(this->_data, tmp._data);
+        std::swap(this->_capacity, tmp._capacity);
+        return *this;
+    }
+    // thread safe
+    void clear();
+    jobject handle() { return _handle; }
+    void* data() { return _data; }
+    int capacity() { return _capacity; }
+
+private:
+    jobject _handle;
+    void* _data;
+    int _capacity;
```

Comment:
Need init.

Reply:
We don't provide a default constructor, so there's no need to add a default initialization value to the other properties